.github/workflows/nightly.yml

```diff
@@ -0,0 +1,71 @@
+name: Publish Nightly Builds
+
+on:
+  push:
+    branches:
+      - nightly_build
+
+jobs:
+  build-test-publish:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - runson: macos-13 # mac x64
+            archive: async-profiler-3.0-macos-x64.zip
+
+          - runson: macos-14 # mac arm
+            archive: async-profiler-3.0-macos-arm64.zip
+
+          - runson: codebuild-test-x86-${{ github.run_id }}-${{ github.run_attempt }}
+            archive: async-profiler-3.0-linux-x64.tar.gz
+
+          - runson: codebuild-test-${{ github.run_id }}-${{ github.run_attempt }}
+            archive: async-profiler-3.0-linux-arm64.tar.gz
+    runs-on: ${{ matrix.runson }}
+    steps:
+      - uses: actions/setup-java@v4
+        with:
+          distribution: 'corretto'
+          java-version: '11'
+      - uses: actions/checkout@v4
+      - name: Build and Test
+        run: |
+          case "${{ matrix.runson }}" in
+            macos*) make release test
+            ;;
+            *) make CC=/usr/local/musl/bin/musl-gcc release test
+            ;;
+          esac
+      - uses: actions/upload-artifact@v4
+        with:
+          name: ${{ matrix.archive }}-${{ github.sha }}
+          path: ${{ matrix.archive }}
+      - uses: actions/upload-artifact@v4
+        if: always()
```

Comment:
What's the purpose of `if: always()`?

Reply:
it makes the step always run, so if the test fails the logs will still be uploaded